test/customValidators_test.js

```diff
@@ -831,7 +831,77 @@ describe('validateChannel()', function () {
       validateChannels(parsedInput, inputString, input);
     }, expectedErrorObject);
   });
+
+  it('should successfully validate channel with servers that are all declared in servers object', async function () {
+    const inputString = `{
+      "servers": {
+        "server1": {
+          "url": "http://localhost",
+          "protocol": "kafka"
+        },
+        "server2": {
+          "url": "http://test.org",
+          "protocol": "jms"
+        }
+      },
+      "channels": {
+        "jmsQueue": {
+          "servers": ["server2"]
+        }
+      }
+    }`;
+    const parsedInput = JSON.parse(inputString);
+
+    expect(validateChannels(parsedInput, inputString, input)).to.equal(true);
+  });
+
+  it('should throw error that servers list references an unknown server', async function () {
+    const inputString = `{
+      "servers": {
+        "server1": {
+          "url": "http://localhost",
+          "protocol": "kafka"
+        },
+        "server2": {
+          "url": "http://test.org",
+          "protocol": "jms"
+        }
+      },
+      "channels": {
+        "jmsQueue": {
+          "servers": ["server2", "unknownServer"]
+        }
+      }
+    }`;
+    const parsedInput = JSON.parse(inputString);
+
+    const expectedErrorObject = {
+      type: 'https://github.com/asyncapi/parser-js/validation-errors',
+      title: 'Channel validation failed',
+      parsedJSON: parsedInput,
+      validationErrors: [
+        {
+          title:
+            'jmsQueue channel is defined for servers that are not defined in the document: unknownServer',
```

Comment:
the final message is confusing, it should  say something like `jmsQueue channel contains servers that are not on the main servers list in the root of the document: unknownServer`

Reply:
I would prefer without the word `main`:

```
jmsQueue channel contains servers that are not on the servers list in the root of the document: unknownServer
```

Reply:
@derberg I added this test locally but I don't have permission to push changes. Can we merge the current PR and I will quickly add a follwup PR? Gerald wrote that he would not have much time until the end of the week.